### DISCUSSION TITLE
Is it possible to use Radix without javascript/typescript?

### DISCUSSION THREAD

**Discussion by opentyler** [radix-ui/themes#197] [General]
I try to avoid as much javascript as possible and have sane html/css defaults. I'm just wondering how hard it would be to build on top of Radix components but tweak them to have better defaults when javascript is disabled. For example, the scroll area on the primitives page example, when javascript is disabled, there is no ability to scroll at all; whereas ideally it would default to a sensible html and css that allows the browser to use sensible default behaviour. I know the javascript is very helpful for ensuring a consistent expierence across browsers and for certain accessibility concerns, but how hard would it be to try and achieve more graceful degradation when javascript is not enabled? Otherwise Radix looks great!!

**Comment by vladmoroz** [comment]
Components that aren't interactive, like layout components, Card, Callout, Text, etc, or components that rely only on the native browser features like Button and Link work fine when pre-rendered into HTML.

Everything else that is built on top of the primitives, like Dropdown Menu or Dialog, won't work without JavaScript.

There is almost nothing you could do about it on your side. I'll close this discussion as it's something we wouldn't ever prioritise – it's a pretty niche request for a React lib.
